File: backend/app/Models/Vehicle.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;

class Vehicle extends Model
{
    use HasFactory;

    protected $fillable = [
        'image_url',
        'type',
        'model',
        'color',
        'registration_number',
        'capacity',
    ];

    // Vehicle belongs to delivery personnel
    public function personnel()
    {
        return $this->hasOne(DeliveryPersonnel::class);
    }

    // Vehicle can have one driver
    public function driver()
    {
        return $this->hasOne(Driver::class);
    }
}

File: frontend/lib/features/go_ride/data/models/vehicle.dart
import 'package:equatable/equatable.dart';

class Vehicle extends Equatable {
  final int id;
  final String imageUrl;
  final String type;
  final String model;
  final String color;
  final String registrationNumber;
  final int capacity;

  const Vehicle({
    required this.id,
    required this.imageUrl,
    required this.type,
    required this.model,
    required this.color,
    required this.registrationNumber,
    required this.capacity,
  });

  factory Vehicle.fromJson(Map<String, dynamic> json) {
    return Vehicle(
      id: json['id'],
      imageUrl: json['image_url'],
      type: json['type'],
      model: json['model'],
      color: json['color'],
      registrationNumber: json['registration_number'],
      capacity: json['capacity'],
    );
  }

  Map<String, dynamic> toJson() => {
        'id': id,
        'image_url': imageUrl,
        'type': type,
        'model': model,
        'color': color,
        'registration_number': registrationNumber,
        'capacity': capacity,
      };

  @override
  List<Object?> get props =>
      [id, imageUrl, type, model, color, registrationNumber, capacity];
}
